src/components/OwnerMessage/OwnerMessage.jsx

```diff
@@ -127,7 +127,7 @@ function OwnerMessage({
     <div className="message-container">
       {ownerMessage ? getContent(ownerMessage) : getContent(ownerStandardMessage)}
 
-      {user.role === 'Owner' && (
+      {(props.hasPermission('editHeaderMessage')) && (
```

Comment:
`props` is not defined here. 2 things:
- Add `...props` in the component definition should do the job.
- Under `mapDispatchToProps` function, add
```diff
const mapDispatchToProps = dispatch => ({
  getMessage: () => dispatch(getOwnerMessage()),
  updateMessage: ownerMessage => dispatch(updateOwnerMessage(ownerMessage)),
  deleteMessage: () => dispatch(deleteOwnerMessage()),
+  hasPermission: permission => dispatch(hasPermission(permission)),
});
```